### PR TITLE
feat(core-ordered-list): upgrade to styled-components

### PR DESCRIPTION
**Pull request by jraff**
BREAKING CHANGE: upgrade to styled-components

## Related issues

Needs https://github.com/telus/tds-core/pull/1037/files#diff-f4dd02fdb3b955c7c26ff698a1f802b2 from the unordered list styled component rewrite.

## Checklist before submitting pull request

- [ ] New code is unit tested
- Commits follow our [Developer Guide](https://tds.telus.com/contributing/developer-guide.html#make-a-commit)
- [ ] For code changes, run `npm run prepr` locally
  - make sure visual and accessibility tests pass

### CODE REVIEW

**Comment by marcod1419** [comment]
This is now ready for review.

File: packages/OrderedList/OrderedList.jsx [hunk 1]
```diff
@@ -56,3 +61,5 @@ OrderedList.defaultProps = {
 OrderedList.Item = OrderedItem
 
 export default OrderedList
+
+export { StyledOrderedItem }
```
Comment (theetrain) [inline packages/OrderedList/OrderedList.jsx]: Should we instead export this within OrderedItem.jsx instead?

File: packages/OrderedList/OrderedList.jsx [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
+  lowerAlpha: 'lower-alpha',
+  decimal: 'decimal',
+}
 
-const injectListStyle = (child, size) => React.cloneElement(child, { size })
+export const StyledOrderedList = styled(({ size, listStyle, ...rest }) => <Box {...rest} />)(
+  ({ listStyle }) => ({
+    paddingLeft: '3rem',
+    listStyleType: listStyleType[listStyle],
+    ...list.nestedListSpacing,
```
Comment (theetrain) [inline packages/OrderedList/OrderedList.jsx]: Please double check the implementation of this.

image

**Comment by marcod1419** [comment]
> The latest sensation is coming to a node registry near you. Critics are raving about OrderedList 3.0.0
> 
> > This time, it's in order.
> > -- Rolling Numbers Magazine
> 
> > All your friends will want to write lists.
> > -- SnozzFeed
> 
> Now with more:
> 
> * Less unit test warnings
> * Mergability
> * Ghost e2e snapshot failures
> * **But seriously, how come step tracker doesn't like my local Docker container?**

This is the `#1` family styled component of the summer! Experience the heartwarming tale of a list that tries to keep things in order and finds out what family really means. Several `OrderedList.Item`s.

**Pull request merged**